source/extensions/common/wasm/context.cc

```diff
@@ -1723,6 +1723,9 @@ Http::FilterDataStatus Context::decodeData(::Envoy::Buffer::Instance& data, bool
   if (!in_vm_context_created_) {
     return Http::FilterDataStatus::Continue;
   }
+  if (buffering_request_body_) {
+    decoder_callbacks_->addDecodedData(data, false);
+  }
```

Comment:
I think this will make the final output contains repeated data piece?

/wait

Reply:
I wasn't familiar with that function. and when try to understand where it was used and for what, I found this comment 
https://github.com/envoyproxy/envoy/blob/main/source/server/admin/admin_filter.cc#L25
in our case streamming is a supported scenario and that is the reason it is only dumped in buffering scenario


I think I have that scenario covered with the test I have added. the one add at the end of the body '.0' 
if you want me to add other test I could.
I made the suite for http and http/2 because this issue is not happening in http1, it seams the library is always sending the last chunk containing end_of_stream with 0 bytes.

Reply:
the addDecodedData use the move semantics. So, yeah, then this should be safe. I will take a more detailed look this night.